src/main/java/uk/gov/companieshouse/officerfiling/api/controller/OfficerControllerImpl.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.companieshouse.officerfiling.api.controller;
+
+import static uk.gov.companieshouse.officerfiling.api.utils.Constants.TRANSACTION_ID_KEY;
+
+import java.util.HashMap;
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestAttribute;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.officerfiling.api.exception.OfficerServiceException;
+import uk.gov.companieshouse.officerfiling.api.service.OfficerService;
+import uk.gov.companieshouse.officerfiling.api.utils.LogHelper.Builder;
+
+@RestController
+public class OfficerControllerImpl implements OfficerController {
+
+    private final OfficerService officerService;
+    private final Logger logger;
+
+    public OfficerControllerImpl(final OfficerService officerService, final Logger logger) {
+        this.officerService = officerService;
+        this.logger = logger;
+    }
+
+    @Override
+    @ResponseBody
+    @GetMapping(value = "/transactions/{transactionId}/officers/active-officers-details", produces = {"application/json"})
```

Comment:
Should this be marked as a private endpoint? I don't imagine third party filers would need to access it.

Reply:
Also we'd need to make sure it's protected from external access. Marking it as private will require an internal CH key to be used for auth.